### PR TITLE
SignUP - Target -signup page created

### PR DESCRIPTION
**Pull request by mwtz**
## Links:

[Ticket](https://trello.com/c/gJwx5EZi/2-5-as-a-user-i-should-be-able-to-sign-up-so-my-account-is-created-in-target-mvd)

## What & Why:

Layout for Sign up in Target MVD to create an new account

### Features:
* Name and gender fields added to the form with their corresponding validations
* Figma design implementation
* API call
* Creation of the mobile version component

image

- [ ] Safe to Revert *check this box if this PR can be reverted without incident*

### CODE REVIEW

File: src/components/form/select/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { string, func } from 'prop-types';
+import cn from 'classnames';
+import './styles.css';
+
+const Select = ({ register, options, name, error, placeholder, className = '', ...rest }) => (
+  <div className="Select">
```
Comment (diebarral) [inline src/components/form/select/index.js]: Try to keep class names uncapitalized.

File: src/components/form/select/styles.css [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.Select select {
```
Comment (diebarral) [inline src/components/form/select/styles.css]: Maybe it's a good idea to start using sass, it helps you write cleaner css code.

File: src/components/mobileSample/styles.css [hunk 3]
```diff
@@ -0,0 +1,23 @@
+.mobile-container{
+    background-color: var(--color-light-blue);
+    color: var(--color-light-black);
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    height: 100vh;
+    width: 50vw;
```
Comment (diebarral) [inline src/components/mobileSample/styles.css]: Remember to keep alphabetical order on the styles properties so it's easier to find them later.

File: src/index.css [hunk 4]
```diff
@@ -1,7 +1,8 @@
+@import url('https://fonts.googleapis.com/css2?family=Open+Sans:ital,wght@0,300;0,400;0,500;0,600;1,300;1,400;1,500&display=swap');
+
 body {
   margin: 0;
-  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu',
-    'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
+  font-family: 'Open Sans', sans-serif;
```
Comment (diebarral) [inline src/index.css:5]: You can do this in `src/index.css` file, so it affects the whole app and not just the forms.

File: src/pages/Signup/index.js [hunk 5]
```diff
@@ -61,47 +68,73 @@ const Signup = () => {
...
+            type="password"
+            name="passwordConfirmation"
+            error={errors.passwordConfirmation}
+            handleFocus={handleFocus}
+          />
+
+          <label htmlFor="gender">{t('signup.labels.gender')}</label>
+          <Select
+            name="gender"
+            options={['female', 'male', 'other']}
```
Comment (diebarral) [inline src/pages/Signup/index.js]: Maybe you could move these options to a constant, or even better, to a constants file. This way if you need to reuse them somewhere else (even though in this case is unlikely), you don't have to redefine them.

File: src/pages/Signup/index.js [hunk 6]
```diff
@@ -61,47 +68,73 @@ const Signup = () => {
...
+          />
+
+          <label htmlFor="password">{t('signup.labels.password')}</label>
+          <Input
+            register={register}
+            type="password"
+            name="password"
+            error={errors.password}
+            handleFocus={handleFocus}
+            placeholder="MIN. 6 CHARACTERS LONG"
```
Comment (diebarral) [inline src/pages/Signup/index.js]: Maybe you can keep this and other placeholder texts in `locales/en.json`

File: src/index.css (hunk 4 above)
Comment (mwtz) [reply src/index.css:5]: This change is in. `src/index.css`

Comment (diebarral) [reply src/index.css:5]: Oh, I must've confused the files or something, sorry about that.

**Review by diebarral — APPROVED** [review]

File: src/components/form/select/index.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { string, func } from 'prop-types';
+import cn from 'classnames';
+import './styles.css';
+
+const Select = ({ register, options, name, error, placeholder, className = '', ...rest }) => (
```
Comment (dariokozicki) [inline src/components/form/select/index.js:6]: Consider using a UI library for these common components. Later on you'll want to customize the option items, for example. Good options are PrimeReact, Material UI, Bootstrap.
image

File: src/components/form/Input/styles.css [hunk 8]
```diff
@@ -1,16 +1,12 @@
-.Input {
-  margin: 0.3rem 0 0.8rem;
-}
-
 .Input input {
-  border-radius: 5px;
-  border: 1px solid var(--color-black);
+  border: 1px solid var(--color-light-black);
```
Comment (dariokozicki) [inline src/components/form/Input/styles.css]: Consider installing Sass for the usage of variables and mixins.

File: src/components/mobileSample/index.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import './styles.css';
```
Comment (dariokozicki) [inline src/components/mobileSample/index.js]: Order imports, styles usually go at the end.

File: src/pages/Signup/index.js [hunk 10]
```diff
@@ -42,7 +46,10 @@ const Signup = () => {
     formState: { errors },
   } = useForm({ resolver: zodResolver(schema) });
 
-  const onSubmit = data => signup(data);
+  const onSubmit = data => {
+    console.log(data);
```
Comment (dariokozicki) [inline src/pages/Signup/index.js]: Avoid commiting console logs

File: src/components/form/select/index.js (hunk 7 above)
Comment (mwtz) [reply src/components/form/select/index.js:6]: Great, yes we talked about this, next tickets will be implementing with some library

File: src/components/form/Input/styles.css (hunk 8 above)
Comment (mwtz) [reply src/components/form/Input/styles.css]: Ok, I already installed Sass

**Review by dariokozicki — APPROVED** [review]

**Pull request merged**